### ISSUE TITLE
Fails with just poetry-core

### ISSUE THREAD

**Issue by Yajo** [mtkennerly/poetry-dynamic-versioning#35]
For some reason, doc builds in readthedocs for https://github.com/copier-org/copier started failing recently.

You can see https://readthedocs.org/projects/copier/builds/12592301/ for example.

[The relevant configuration in `pyproject.toml`](https://github.com/copier-org/copier/blob/679fd7cd8471af5434d17590717efcd47dd93ccf/pyproject.toml#L110-L112) was:

```toml
[build-system]
requires = ["poetry_core>=1.0.0", "poetry-dynamic-versioning"]
build-backend = "poetry.core.masonry.api"
```

The error was:

```
/home/docs/checkouts/readthedocs.org/user_builds/copier/envs/317/bin/python -m pip install --upgrade --upgrade-strategy eager --no-cache-dir .[docs]
Processing /home/docs/checkouts/readthedocs.org/user_builds/copier/checkouts/317
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
    Preparing wheel metadata: started
    Preparing wheel metadata: finished with status 'error'
    ERROR: Command errored out with exit status 1:
     command: /home/docs/checkouts/readthedocs.org/user_builds/copier/envs/317/bin/python /home/docs/checkouts/readthedocs.org/user_builds/copier/envs/317/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpcl96uht6
         cwd: /tmp/pip-req-build-_madcsdq
    Complete output (12 lines):
    Traceback (most recent call last):
      File "/home/docs/checkouts/readthedocs.org/user_builds/copier/envs/317/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 280, in <module>
        main()
      File "/home/docs/checkouts/readthedocs.org/user_builds/copier/envs/317/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 263, in main
        json_out['return_val'] = hook(**hook_input['kwargs'])
      File "/home/docs/checkouts/readthedocs.org/user_builds/copier/envs/317/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py", line 133, in prepare_metadata_for_build_wheel
        return hook(metadata_directory, config_settings)
      File "/tmp/pip-build-env-ze9az9mi/overlay/lib/python3.8/site-packages/poetry/core/masonry/api.py", line 34, in prepare_metadata_for_build_wheel
        poetry = Factory().create_poetry(Path(".").resolve())
      File "/tmp/pip-build-env-ze9az9mi/overlay/lib/python3.8/site-packages/poetry_dynamic_versioning/__init__.py", line 287, in alt_poetry_create
        run_mod = _state.original_import_func("poetry.console.commands.run", fromlist=[None])
    ModuleNotFoundError: No module named 'poetry.console'
    ----------------------------------------
ERROR: Command errored out with exit status 1: /home/docs/checkouts/readthedocs.org/user_builds/copier/envs/317/bin/python /home/docs/checkouts/readthedocs.org/user_builds/copier/envs/317/lib/python3.8/site-packages/pip/_vendor/pep517/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpcl96uht6 Check the logs for full command output.
```

I have changed it to this, as a workaround to fix the build, but previous config was working fine until recently:

```toml
[build-system]
requires = ["poetry>=1.1.0", "poetry-dynamic-versioning"]
build-backend = "poetry.core.masonry.api"
```

**Comment by mtkennerly** [comment]
I think I see where the issue is in the plugin, although I'm not sure why it would have been working until recently and only just started having an issue, unless there was an environmental change on the Read the Docs servers. How long ago was it working for you?

**Comment by Yajo** [comment]
https://readthedocs.org/projects/copier/builds/12440919/ was the last successful build I can see before I applied my workaround.

**Comment by mtkennerly** [comment]
@Yajo I just released v0.12.1, which should fix this. Please let me know if you run into any other similar issues.
